File: 08. Linked List/SinglyLinkedList.java
public class SinglyLinkedList{
    public static void main(String[] args){
        Node head= new Node(10);
        Node temp1 = new Node(20);
        Node temp2 = new Node(30);
        head.next=temp1;
        temp1.next=temp2;
       // System.out.println(head.data);
        //System.out.println(temp1.data);
        //System.out.println(temp2.data);
        System.out.println(head.data+ " "+temp1.data+" " +temp2.data);
        
    }
    
}
class Node{
    int data;
    Node next;
    Node(int a){
        data =a;
        next = null;
    }


}

File: 08. Linked List/Singly_Linked_List.cpp
 #include<iostream>
using namespace std;
class node{
    public:
    int data;
    node* next;
    node(int data){
        this->data=data;
        next=NULL;
    }
};
node* takeinput(){
        int data;
        cin>>data;                             
        node*head=NULL;
        node*tail=NULL;

        while(data!=-1){
            node* newnode= new node(data);
         if(head==NULL){                                           
            head=newnode;
            tail=newnode;

        }
        else{
            tail->next=newnode;
            tail=tail->next;}
        cin>>data;}
        return head;}
        
      node* insertnode(node *head,int i,int data){
        node*temp=head;
        node*newnode=new node(data);
        int count=0;
        if(i==0){
            newnode->next=head;
            head=newnode;
            return head;      
        }                        
      while(temp!=NULL && count<i-1){
        temp=temp->next;
        count++;
        }
        if(temp!=NULL){
           newnode->next=temp->next;
           temp->next=newnode;
        }
        return head;
       };
      
        void reverseList(node*head){
           if(head->next!=NULL){
            reverseList(head->next);
             }  
           cout<<head->data<<" ";
           return;
           
        };
        void print(node*head){node*temp=head;
            while(temp!=NULL){
                cout<<temp->data<<" " ;
                temp=temp->next;}
                cout<<endl;
        }
        int main(){
             node*head=takeinput();
             insertnode(head,3,99);
             print(head);
             reverseList(head);


 }